### DISCUSSION TITLE
Fit linear structural causal model to causal parents taken from graph

### DISCUSSION THREAD

**Discussion by robinsonhordoir** [jakobrunge/tigramite#302] [Q&A]
Dear all,

I am totally new to Tigramite, and I'm trying to apply this test case :
https://github.com/jakobrunge/tigramite/blob/master/tutorials/benchmarking_and_validation/tigramite_tutorial_explaining_correlations.ipynb

to my own datasets, and so far I managed to follow the steps to create the causal graph and the results make sense.
However, I lack the expertise the re-create the signal from the causal parents taken from graph as specified in point 5, where it is just
specified "Using the Prediction class which calls Models().", which does not mean anything obvious for me.

Could anyone provide me with an example ?

Thanks a lot in advance for your help,

Robinson

**Comment by jakobrunge** [comment]
These steps are all wrapped in the function generate_linear_model_from_data from the module toymodels.surrogate_generator. So it is done internally. You don't have to do it yourself.

**Reply by robinsonhordoir** [reply]
Hei Jakob,

Thanks for the quick reply, I wrote the following line:
lm= generate_linear_model_from_data(dataframe, parents, tau_max)

but where do I find the reconstructed variables ?

Thanks for your help,

Robinson

**Comment by jakobrunge** [comment]
You can learn these as part of causal discovery. See https://github.com/jakobrunge/tigramite/blob/master/tutorials/causal_discovery/tigramite_tutorial_causal_discovery_overview.ipynb
